Gravity-tube/src/Pages/AuthWidget/Login/Login.js

```diff
@@ -0,0 +1,185 @@
+import "./Login.css";
+import { Link } from "react-router-dom";
+import { InputPwd } from "../InputPwd";
+import { useNavigate } from "react-router-dom";
+import { useReducer } from "react";
+import axios from "axios";
+import { useAuth } from "../../../Contexts/Auth-context";
+import { ToastContainer, toast } from "react-toastify";
+import 'react-toastify/dist/ReactToastify.css';
+
+export function Login() {
+  const { authState, authDispatch } = useAuth();
+
+  const navigate = useNavigate();
+
+  const initialState = {
+    email: "",
+    password: "",
+  };
+
+  const dummyData = {
+    email: "guest@gmail.com",
+    password: "Abc@1234",
+  };
+
+  const formReducer = (state, { type, payload }) => {
```

Comment:
put this reducer function to a different file.

Reply:
yeah sure